bescort.lic

```diff
@@ -934,6 +940,32 @@ class Bescort
       move 'out'
     end
   end
+
+  def take_xing_ferry(mode)
+    unless [1904, 957].include?(Room.current.id)
+      echo "You are not at the ferry docks"
+      return
+    end
+
+    if wealth('Crossing') < 35
+      echo('Get money you slob!')
+      return
+    end
+
+    pause 1 until DRRoom.room_objs.find{ |x| x =~ /the ferry/ }
+
+    case bput('go ferry', 'You hand him', 'The ferry has just pulled away from the dock', 'There is no ferry here to go aboard', 'Come back when you can afford the fare')
+    when 'You hand him'
+      waitfor 'reaches the dock and its crew ties the ferry off'
+      move 'go dock'
+    when 'The ferry has just pulled away from the dock', 'There is no ferry here to go aboard'
+      pause 1 while DRRoom.room_objs.find{ |x| x =~ /the ferry/ }
```

Comment:
When the ferry pulls away from the dock, it still exists in the `room_objs` array for a few seconds. This is just to keep the script from spamming 'go ferry'